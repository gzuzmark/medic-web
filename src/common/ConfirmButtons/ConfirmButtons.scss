.ConfirmButtons {
  display: flex;
  justify-content: space-around;
  margin-top: 85px;
  margin-bottom: 20px;
  &-button {
    display: inline-block;
    width: 256px;
    &:first-child {
      margin-right: 40px;
    }
  }
}