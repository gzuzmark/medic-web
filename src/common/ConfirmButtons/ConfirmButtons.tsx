import * as React from 'react';
import './ConfirmButtons.scss';

interface IPropsConfirmButtons {
    styles?: React.CSSProperties;
    disabled: boolean;
    loading: boolean;
    cancelText: string;
    confirmText: string;
    onCancel(): void;
    onConfirm(): void;
}

const ConfirmButtons: React.FC<IPropsConfirmButtons> = (props) => {
    return (
        <div className="ConfirmButtons" style={{...props.styles}}>
            <button className="u-Button u-Button--white ConfirmButtons-button"
                    onClick={props.onCancel}>{props.cancelText}</button>
            <button className="u-Button ConfirmButtons-button"
                    disabled={props.disabled}
                    onClick={props.onConfirm}
                    data-loading={props.loading ? true : undefined}>
                {props.loading ? '' : props.confirmText}</button>
        </div>
    );
};


export default ConfirmButtons;
