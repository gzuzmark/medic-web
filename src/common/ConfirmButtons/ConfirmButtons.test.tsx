import { mount } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import ConfirmButtons from './ConfirmButtons';

describe('ConfirmButtons Test',() => {
    let props: any;
    let mountedConfirmButtons: any;
    const getComponent = () => {
        if (!mountedConfirmButtons) {
            mountedConfirmButtons = mount(
                <ConfirmButtons {...props} />
            );
        }
        return mountedConfirmButtons;
    };

    beforeEach(() => {
        props = {
            cancelText: "Cancelar",
            confirmText: "Confirmar",
            disabled: false,
            loading: false,
            onCancel: () => {
                //
            },
            onConfirm: () => {
                //
            },
        };
        mountedConfirmButtons = undefined;
    });

    it("render: render ConfirmButtons", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });

    it("render: click on cancel button", () => {
        const click = jasmine.createSpy('click');
        const newProps = {...props};
        newProps.onCancel = click;
        props = newProps;
        const component = getComponent();
        component.find('.ConfirmButtons-button').first().simulate('click');
        expect(click).toHaveBeenCalled();
    });

    it("render: click on confirm button", () => {
        const click = jasmine.createSpy('click');
        const newProps = {...props};
        newProps.onConfirm = click;
        props = newProps;
        const component = getComponent();
        component.find('.ConfirmButtons-button').at(1).simulate('click');
        expect(click).toHaveBeenCalled();
    });

});
