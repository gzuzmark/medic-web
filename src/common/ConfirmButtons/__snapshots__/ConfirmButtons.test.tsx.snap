// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmButtons Test render: render ConfirmButtons 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmButtons
    cancelText="Cancelar"
    confirmText="Confirmar"
    disabled={false}
    loading={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="u-Button u-Button--white ConfirmButtons-button"
          onClick={[Function]}
        >
          Cancelar
        </button>,
        <button
          className="u-Button ConfirmButtons-button"
          data-loading={undefined}
          disabled={false}
          onClick={[Function]}
        >
          Confirmar
        </button>,
      ],
      "className": "ConfirmButtons",
      "style": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cancelar",
          "className": "u-Button u-Button--white ConfirmButtons-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Cancelar",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Confirmar",
          "className": "u-Button ConfirmButtons-button",
          "data-loading": undefined,
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Confirmar",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="u-Button u-Button--white ConfirmButtons-button"
            onClick={[Function]}
          >
            Cancelar
          </button>,
          <button
            className="u-Button ConfirmButtons-button"
            data-loading={undefined}
            disabled={false}
            onClick={[Function]}
          >
            Confirmar
          </button>,
        ],
        "className": "ConfirmButtons",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancelar",
            "className": "u-Button u-Button--white ConfirmButtons-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Cancelar",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Confirmar",
            "className": "u-Button ConfirmButtons-button",
            "data-loading": undefined,
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Confirmar",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
